src/table/table-role/__tests__/utils.tsx

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useState } from 'react';
+import { render } from '@testing-library/react';
+import { GridNavigationContext } from '../../../../lib/components/table/table-role/grid-navigation-context';
+
+function FakeGridNavigationProvider({ target, children }: { target: null | string; children: React.ReactNode }) {
+  const [focusTarget, setFocusTarget] = useState<null | Element>(null);
+  useEffect(() => {
+    if (!target) {
+      setFocusTarget(null);
+    } else {
+      setFocusTarget(document.querySelector(target));
+    }
+  }, [target]);
```

Comment:
This looks like too much complexity for a mock. Does it really need that much to represent the real code?

Reply:
The intention is to provide the expected state to the context which is a boolean flag and an html element. The challenge with the element is that it needs to be obtained. The mock can be made simpler by accepting the element as a prop but then using the mock is going to be more challenging.

Reply:
An alternative option

```js
let setCurrentTarget;
render(<FakeGridNavigationProvider ref={el => setCurrentTarget = el}><Button /></FakeGridNavigationProvider>)

setCurrentTarget(wrapper.findButton().getElement());
```

what pros and cons do you see here and there?